website/docs/r/codeartifact_domain.html.markdown

```diff
@@ -0,0 +1,50 @@
+---
+subcategory: "CodeArtifact"
+layout: "aws"
+page_title: "AWS: aws_codeartifact_domain"
+description: |-
+  Provides a CodeArtifact Domain resource.
+---
+
+# Resource: aws_codeartifact_domain
+
+Provides a CodeArtifact Domains Resource.
+
+## Example Usage
+
+```hcl
+resource "aws_kms_key" "example" {
+  description = "domain key"
+}
+
+resource "aws_codeartifact_domain" "example" {
+  domain         = "example.com"
+  encryption_key = "${aws_kms_key.example.arn}"
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `domain` - (Required) The name of the domain to create. All domain names in an AWS Region that are in the same AWS account must be unique. The domain name is used as the prefix in DNS hostnames. Do not use sensitive information in a domain name because it is publicly discoverable.
+* `encryption_key` - (Required) The encryption key for the domain. This is used to encrypt content stored in a domain. An encryption key can be a key ID, a key Amazon Resource Name (ARN), a key alias, or a key alias ARN.
```

Comment:
See above for question with validation of this field. If ID or alias (ARN) is supplied, do we have to worry about diffs forcing resource creation?

Reply:
yes, good catch. it does cause diffs